app/models/post.rb

```diff
@@ -57,6 +57,34 @@ def address
     [street, city, state, postal_code, country].compact.join(", ")
   end
 
+  def self.filter_type(type)
+    if type == "needed" || type == "available"
+      send(type.to_sym)
+    else
+      all
```

Comment:
for some reason I'm not able to find docs for an 'all' method anywhere. I can infer what it's doing, but is this a builtin for rails?